.pre-commit-config.yaml

```
@@ -37,7 +37,8 @@ repos:
       # - id: fix-encoding-pragma  # covered by pyupgrade
       - id: trailing-whitespace
         # Trims trailing whitespace.
-        exclude: ".*(data.*|extern.*|licenses.*|_parsetab.py|test_cds.py)$"
+        exclude_types: [python]
```

There should also be a comment explaining why we are doing this. It is obvious in the context of this pull request (so obvious that we didn't even think of this before), but without that context it would look rather strange.
```suggestion
        exclude_types: [python]  # Covered by Ruff W291.
```